Formula/heroku.rb

```diff
@@ -1,17 +1,19 @@
+require "language/node"
+
 class Heroku < Formula
   desc "Command-line client for the cloud PaaS"
   homepage "https://cli.heroku.com"
-  url "https://cli-assets.heroku.com/heroku-cli/channels/stable/heroku-cli-v6.12.3-fca882c-darwin-x64.tar.xz"
-  version "6.12.3"
-  sha256 "926462a2086bde11dcfc940157bd91de6376e9c2cdc6bdade6a9143b205a8886"
-
-  bottle :unneeded
+  url "https://registry.npmjs.org/heroku-cli/-/heroku-cli-6.12.4.tgz"
+  sha256 "0320f25bf786c851805472d352cc8fa6f318ddd11ad3d65e0b1ef9110208c6cd"
+  head "https://github.com/heroku/cli.git"
 
-  depends_on :arch => :x86_64
+  depends_on "node"
 
   def install
-    libexec.install Dir["*"]
-    bin.install_symlink libexec/"bin/heroku"
+    system "npm", "install", *Language::Node.std_npm_install_args(libexec)
+    bin.install_symlink Dir["#{libexec}/bin/*"]
+    inreplace libexec/"lib/node_modules/heroku-cli/bin/run.js", /npm update -g heroku-cli/, "brew upgrade heroku"
+    inreplace libexec/"lib/node_modules/heroku-cli/bin/run", %r{^node "\$DIR\/run\.js"}, "#{Formula["node"].opt_bin}/node \"$DIR/run.js\""
```

Comment:
No I mean for the before part of the replacement. It can be a substring instead of a regex.

`'node "$DIR/run.js"'` should "just work"